src/window/message_dialog.c

```diff
@@ -454,7 +459,8 @@ static void draw_foreground_normal(void)
 
     if (msg->type == TYPE_MESSAGE) {
         image_buttons_draw(data.x + 16, data.y + 16 * msg->height_blocks - 40, get_advisor_button(), 1);
-        if (msg->message_type == MESSAGE_TYPE_DISASTER || msg->message_type == MESSAGE_TYPE_INVASION) {
+        //FIXME: Use is_problem_message and introduce is_event_message
+        if (msg->message_type == MESSAGE_TYPE_DISASTER || msg->message_type == MESSAGE_TYPE_INVASION || msg->message_type == MESSAGE_TYPE_EVENT) {
```

Comment:
I would like to unify these checks, please suggest whether I should use `is_problem_message` or whether I should continue with separate type checks.

Reply:
Humm. Now that I think of it, invasions and disasters are also events.

Maybe change `MESSAGE_TYPE_EVENT` to `MESSAGE_TYPE_BUILDING_COMPLETION` and then create an `is_event_message` that checks all three cases.

Reply:
I will have a look, thanks!